app/backend/src/clj/teet/activity/activity_db.clj

```diff
@@ -21,3 +22,100 @@
               (and (not (.before date (:activity/estimated-start-date activity-dates)))
                    (not (.after date (:activity/estimated-end-date activity-dates)))))
             dates)))
+
+(defn- manager-transactions [db activity-ids]
+  (->> (d/q '[:find ?a ?modified-at ?tx ?ref
+              :in $ [?a ...]
+              :where
+              [?a :activity/manager ?ref ?tx true]
+              [?a :meta/modified-at ?modified-at ?tx true]]
+            (d/history db)
+            activity-ids)
+       (map (fn [[a modified-at tx ref]]
+              {:activity a
+               :modified-at modified-at
+               :tx tx
+               :ref ref}))))
+
+(defn- manager-period [[a b]]
+  (if (nil? b)
+    {:manager (:ref a)
+     :period [(:modified-at a) nil]}
+    {:manager (:ref a)
+     :period [(:modified-at a) (:modified-at b)]}))
+
+(defn- manager-transactions->manager-periods [manager-transactions]
+  (->> manager-transactions
+       (partition 2 1 nil)
+       (map manager-period)))
+
+(defn- users-by-id [db db-ids]
+  (->> (d/q '[:find (pull ?u [:db/id :user/given-name :user/family-name])
+              :in $ [?u ...]]
+            db db-ids)
+       (mapcat (comp (juxt :db/id identity)
+                     first))
+       (apply hash-map)))
+
+
+(defn manager-histories-by-activity [manager-transactions users-by-id]
+  (->> manager-transactions
+       (group-by :activity)
+       (cu/map-vals (comp (partial map #(update % :manager users-by-id))
+                          manager-transactions->manager-periods
+                          (partial sort-by :modified-at)))))
+
+(defn get-activity-ids-of-project
```

Comment:
is get verb necessary, this looks like a pure function.

It could be called `project-activity-ids` to reflect the return value instead.

Reply:
It queries the database? Which is unnecessary btw, since the ids could also be gathered from the project passed to update-activity-histories.